Providers/Profile.swift

```diff
@@ -28,23 +24,16 @@ public protocol SyncManager {
     var syncDisplayState: SyncDisplayState? { get }
 
     func hasSyncedHistory() -> Deferred<Maybe<Bool>>
-    func hasSyncedLogins() -> Deferred<Maybe<Bool>>
```

Comment:
Are these mostly functions which were added but never used 🤔

Reply:
I didn't dig into the history, I just know they are currently not used. I suspect they were either added with the intention of later using them or they were at some point used but were never removed when they stopped being used.

Reply:
The `hasSyncedLogins` function looks like it's used in [LoginDetailViewController.swift](https://github.com/mozilla-mobile/firefox-ios/blob/8231a2fe2279b357b64e8bc74278de49618c99b3/Client/Frontend/Settings/LoginDetailViewController.swift#L309) and [LoginListViewController.swift](https://github.com/mozilla-mobile/firefox-ios/blob/6ca9dc075f18131fd640683158cbbd2803b42db3/Client/Frontend/Login%20Management/LoginListViewController.swift#L319).

Reply:
Both cases bypass the Profile and call into the RustLogins object. They probably should call this method instead of exposing the RustLogins object but I don't understand why it was done that way in the first place so might be missing something and that kind of refactor is probably out of scope for a task that's just supposed to be deleting dead code.

Reply:
> Both cases bypass the Profile and call into the RustLogins object. They probably should call this method instead of exposing the RustLogins object but I don't understand why it was done that way in the first place so might be missing something and that kind of refactor is probably out of scope for a task that's just supposed to be deleting dead code.

Oh, sorry--you are correct, I missed the `.logins`. This might simply be a relic from before `RustLogins` when the `hasSyncedLogins` call was backed by a SQL query to browserDB.

Reply:
I refactored this a little to bring back hasSyncedLogins, but it now lives on the Profile instead, so at least those uses cases no longer reference the RustLogins object directly.

Reply:
That was a nice change! And sorry, I meant to approve this after I realized my mistake yesterday and got diverted. LGTM!